apps/frontend/src/pages/projects/projects.tsx

```diff
@@ -1,30 +1,42 @@
 import { PageLayout } from "~/libs/components/components.js";
+import { EMPTY_ARRAY_LENGTH } from "~/libs/constants/constants.js";
 import { DataStatus } from "~/libs/enums/enums.js";
 import {
 	useAppDispatch,
 	useAppSelector,
+	useCallback,
 	useEffect,
 } from "~/libs/hooks/hooks.js";
 import { actions as projectActions } from "~/modules/projects/projects.js";
 
-import { ProjectCard } from "./components/components.js";
+import { ProjectCard, ProjectsSearch } from "./components/components.js";
 import styles from "./styles.module.css";
 
 const Projects = (): JSX.Element => {
 	const dispatch = useAppDispatch();
-
 	const { dataStatus, projects } = useAppSelector(({ projects }) => projects);
 
 	useEffect(() => {
 		void dispatch(projectActions.loadAll());
 	}, [dispatch]);
 
+	const handleSearchChange = useCallback(
+		(value: string) => {
+			void dispatch(projectActions.loadAll(value));
+		},
+		[dispatch],
+	);
+
+	const hasProject = projects.length === EMPTY_ARRAY_LENGTH;
+
 	const isLoading =
-		dataStatus === DataStatus.IDLE || dataStatus === DataStatus.PENDING;
+		dataStatus === DataStatus.IDLE ||
+		(dataStatus === DataStatus.PENDING && hasProject);
```

Comment:
Hm, why do we need to add hasProject here 🤔

Reply:
With fast Internet connection after user searched projects the loader appears instead of projects and creates "blink" effect. However with slow connection there's no loader at all for a couple of seconds and it doesn't feel as an user-friendly experience too. Should I always keep loader visible during pending state?

Reply:
I'm just thinking about initial loading, when we have status pending but no projects yet

Reply:
Loader will be on initial load both with hasProjects and without it